vercel/resource_alias.go

```diff
@@ -0,0 +1,196 @@
+package vercel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/vercel/terraform-provider-vercel/client"
+)
+
+type resourceAliasType struct{}
+
+// GetSchema returns the schema information for an alias resource.
+func (r resourceAliasType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: `
+Provides an alias resource.
+
+An alias allows a deployment to be accessed through a different URL.`,
+		Attributes: map[string]tfsdk.Attribute{
+			"alias": {
+				Description:   "The alias to be set on the deployment. It will become the subdomain of the Vercel project top level domain",
+				Required:      true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
+				Type:          types.StringType,
+			},
+			"deployment_id": {
+				Description:   "The deployment id to alias",
+				Required:      true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
+				Type:          types.StringType,
+			},
+			"team_id": {
+				Description:   "The team or scope id",
+				Optional:      true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
+				Type:          types.StringType,
+			},
+			"uid": {
+				Computed: true,
+				Type:     types.StringType,
+			},
```

Comment:
Looking at all of the API endpoints:

* https://vercel.com/docs/rest-api#endpoints/aliases/assign-an-alias
  * `uid` in response 
* https://vercel.com/docs/rest-api#endpoints/aliases/delete-an-alias
  * `aliasId` in path
* https://vercel.com/docs/rest-api#endpoints/aliases/get-an-alias
  * `idOrAlias` in path
  * `uid` in response
* https://vercel.com/docs/rest-api#endpoints/aliases/list-deployment-aliases
  * `uid` in response
* https://vercel.com/docs/rest-api#endpoints/aliases/list-aliases
  * `uid` in response

It is consistently called `uid` in the responses, so we thought it best to stick with that. Are you sure you want it named differently in the Terraform provider?

Reply:
For the testing framework data_sources have to have an `id` field, and it's a semi-expected pattern that resources have an `id`. 
For projects, we have `id`, Deployments - `id`, Teams - `id`. I think it's bizarre that this would have `uid` as the field name. 

The aliases API is quite an old API in terms of Vercel's history. I suspect that is why it refers to `uid` (and the API shouldn't change for backwards compatibility), but we can expose it in a more consistent way.